plugin/trino-spanner/src/main/java/io/trino/plugin/spanner/SpannerClient.java

```diff
@@ -0,0 +1,660 @@
+        this.config = spannerConfig;
+        this.identifierMapping = identifierMapping;
+    }
+
+    private static RemoteTableName getRemoteTable(ResultSet resultSet)
+            throws SQLException
+    {
+        String schema = resultSet.getString("TABLE_SCHEM");
+        if (schema != null && schema.equals("")) {
+            schema = null;
+        }
+        return new RemoteTableName(
+                Optional.ofNullable(null),
+                Optional.ofNullable(schema),
+                resultSet.getString("TABLE_NAME"));
+    }
+
+    private static ColumnMetadata getPageSinkIdColumn(List<String> otherColumnNames)
+    {
+        // While it's unlikely this column name will collide with client table columns,
+        // guarantee it will not by appending a deterministic suffix to it.
+        String baseColumnName = "trino_page_sink_id";
+        String columnName = baseColumnName;
+        int suffix = 1;
+        while (otherColumnNames.contains(columnName)) {
+            columnName = baseColumnName + "_" + suffix;
+            suffix++;
+        }
+        return new ColumnMetadata(columnName, BigintType.BIGINT);
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int jdbcType = typeHandle.getJdbcType();
+        System.out.println("Column mapping for type " + typeHandle);
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        switch (jdbcType) {
+            case Types.SMALLINT:
+            case Types.INTEGER:
+            case Types.TINYINT:
+            case Types.BIGINT:
+                return Optional.of(StandardColumnMappings.bigintColumnMapping());
+            case Types.NUMERIC:
+            case Types.DECIMAL:
+                return Optional.of(StandardColumnMappings.decimalColumnMapping(DecimalType.createDecimalType(9, 38)));
+            case Types.REAL:
+            case Types.FLOAT:
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.CHAR:
+            case Types.VARCHAR:
+            case Types.NVARCHAR:
+            case Types.LONGVARCHAR:
+            case Types.LONGNVARCHAR:
+                return Optional.of(defaultVarcharColumnMapping(typeHandle.getRequiredColumnSize(), false));
+            case Types.BINARY:
+            case Types.VARBINARY:
+            case Types.LONGVARBINARY:
+                return Optional.of(ColumnMapping.sliceMapping(VARBINARY, varbinaryReadFunction(), varbinaryWriteFunction(), FULL_PUSHDOWN));
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        dateReadFunctionUsingLocalDate(),
+                        spannerDateWriteFunctionUsingLocalDate()));
+            case Types.TIMESTAMP:
```

Comment:
I would recommend removing support for `TIMESTAMP` in this PR. We can add the type later.